src/main/java/teammates/sqllogic/core/FeedbackResponsesLogic.java

```diff
@@ -503,4 +503,27 @@ public void updateFeedbackResponsesForChangingSection(Course course, String newE
         }
     }
 
+    /**
+     * Updates a student's email in their given/received responses.
+     */
+    public void updateFeedbackResponsesForChangingEmail(String courseId, String oldEmail, String newEmail)
+            throws InvalidParametersException, EntityDoesNotExistException {
+
+        List<FeedbackResponse> responsesFromUser =
+                getFeedbackResponsesFromGiverForCourse(courseId, oldEmail);
+
+        for (FeedbackResponse response : responsesFromUser) {
+            response.setGiver(newEmail);
+            frDb.updateFeedbackResponse(response);
```

Comment:
should we also be calling `updateFeedbackResponseCascade` like the original? It says "If the giver/recipient field is changed, the response is updated by recreating the response as question-giver-recipient is the primary key.",

so it seems related

Reply:
I think the javadocs might be outdated? It seems `frDb.updateFeedbackResponse` in the datastore version of `updateFeedbackResponseCascade` does do the recreation, but the current sql version simply merges.